### PR TITLE
Branch + Cross promotion experiment

### PR DESCRIPTION
**Pull request by prashtx**
Switch to branch for our deep linking provider

Implement a cross-promotion interstitial experiment with 3 variants, shown only for direct visits to the front page.

We use client-generated long-form branch links, so we can avoid making 2-3 round trips to the API.

Most of the `shouldShowBanner` logic has been rolled into the feature flag system, so we can control this consistently and centrally. Because we can generate links client-side, we can create them as deterministic functions of state for a component that needs them.

### CODE REVIEW

**Comment by prashtx** [comment]
👓  @uzi @schwers

File: src/app/components/InterstitialPromo/index.jsx [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+      utm_name: 'xpromo_interstitial',
+      utm_content: 'element_1',
+    }),
+    getBranchLink(state, {
+      feature: 'interstitial',
+      campaign: 'xpromo_interstitial',
+      utm_medium: 'interstitial',
+      utm_name: 'xpromo_interstitial',
+      utm_content: 'element_2',
+    }),
```
Comment (uzi) [inline src/app/components/InterstitialPromo/index.jsx]: 🔕  You could clean this up a pinch.

**Comment by uzi** [comment]
Did a quick live review with @prashtx -- looks good to me.  I'll give it another pass tonight.

**Comment by prashtx** [comment]
Added a couple of minor commits and split the experiment into iOS and Android experiments.

:haircut_man:

File: src/app/featureFlags.js [hunk 2]
```diff
@@ -3,24 +3,36 @@ import omitBy from 'lodash/omitBy';
...
+  VARIANT_XPROMO_RATING,
 } = flagConstants;
 
 const config = {
   [BETA]: true,
-  [SMARTBANNER]: true,
+  [SMARTBANNER]: {
+    and: [
+      { allowedPages: ['index', 'listing'] },
+      { allowedDevices: IOS_DEVICES.concat(ANDROID) },
```
Comment (uzi) [inline src/app/featureFlags.js]: 🔕 You also have `RECOGNIZED_DEVICES` which you can import and use.

**Comment by uzi** [comment]
🐟

**Pull request merged**